server/devpi_server/extpypi.py

```diff
@@ -245,6 +254,9 @@ def _save_cache_links(self, project, links, serial):
         if old != data:
             threadlog.debug("saving changed simplelinks for %s: %s", project, data)
             key.set(data)
+            # add project if it isn't already, so deletion and offline mode
```

Comment:
Rephrase comment:

maintain list of currently cached project names to enable deletion and offline mode